upload.go

```diff
@@ -61,9 +64,20 @@ func (u *Upload) Put() {
 
 	relativePath := u.RelativePath()
 
-	err = b.PutReaderHeader(relativePath, file, stat.Size(), headers, s3.ACL("public-read"))
+	attempt := 1
+	for {
+		fmt.Printf("[%d] Path: %s\n", attempt, relativePath)
+
+		err = b.PutReaderHeader(relativePath, file, stat.Size(), headers, s3.ACL("public-read"))
+		if err == nil || attempt >= MaxAttempts {
+			break
```

Comment:
I think you can refactor this further so you don't need to rely on `err` to `log.Fatal` on line 82.